src/main/java/com/google/api/gax/grpc/ApiException.java

```diff
@@ -31,17 +31,24 @@
 
 package com.google.api.gax.grpc;
 
+import com.google.common.base.Preconditions;
+
+import io.grpc.Status;
+
 /**
  * Represents an exception thrown during an RPC call.
  *
  * It stores information useful for functionalities in {@link ApiCallable}.
  */
 public class ApiException extends RuntimeException {
   private final boolean retryable;
+  private final Status.Code statusCode;
 
-  ApiException(Throwable cause, boolean retryable) {
+  ApiException(Throwable cause, Status.Code statusCode, boolean retryable) {
     super(cause);
+    Preconditions.checkNotNull(statusCode);
```

Comment:
checkNotNull actually passes through its value, so you can assign the result of this to this.statusCode.